metrics-operator/controllers/common/providers/dynatrace/common_test.go

```diff
@@ -12,16 +12,12 @@ import (
 )
 
 func TestGetSecret_NoKeyDefined(t *testing.T) {
-	svr := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		_, err := w.Write([]byte(dtpayload))
-		require.Nil(t, err)
-	}))
-	defer svr.Close()
+
```

Comment:
with the check for a key withing the secret ref being removed from the `KeptnProvider.HasSecretDefined()` function, should we update the `getDTSecret` to do an additional check before retrieving the secret? This way we save calls to the k8s API if there is no key defined

Reply:
fair point I will add it